### ISSUE TITLE
Health Check API reports ok even if deployment failed

### ISSUE THREAD

**Issue by baztian** [payara/Payara#2641]
Health Check API reports ok even if deployment failed
----------

The microprofile health check API reports an ok status even if the deployment failed. E.g. I have an app with a @Startup method that raises an exception. The deployment fails but the health api reports ok. (I've not implemented a custom `HealthCheck`)

## Expected Outcome

Health api [should return a HTTP 503](/eclipse/microprofile-health/blob/master/spec/src/main/asciidoc/protocol-wireformat.adoc#status-codes) and the `outcome` attribute should be `DOWN`.

## Current Outcome

HTTP 200 and  `outcome` attribute is `UP`.

## Steps to reproduce

1. Have a JEE application with a `@javax.ejb.Startup` annotation. In the `@PostConstruct` method raise a `RuntimeException`.

2. Start the domain

3. Deploy the application

4. Access http://localhost:8080/health

## Environment ##

- **Payara Version**: 5.181
- **Edition**: Full
- **JDK Version**: 8 - OpenJDK
- **Operating System**: Linux
- **Database**: None

**Comment by mikecroft** [comment]
I've reproduced the reported behaviour easily with this:

```java
import javax.annotation.PostConstruct;
import javax.ejb.Singleton;
import javax.ejb.Startup;

@Singleton
@Startup
public class FailingBean {
    
    @PostConstruct
    public void init(){
        System.out.println("PostConstruct");
        throw new RuntimeException();
    }
}
```

I'm not sure that's the whole story, though. My understanding of the MicroProfile HealthCheck API is that you need to report the health of your applications yourself. So any application which does not have any MicroProfile HealthCheck code in it will be ignored. I tried to verify this myself, however, with an EJB which implements HealthCheck (and does NOT throw the exception) and I found the Health reported on the endpoint did not change. My altered class was:

```java
import javax.annotation.PostConstruct;
import javax.ejb.Singleton;
import javax.ejb.Startup;
import org.eclipse.microprofile.health.HealthCheck;
import org.eclipse.microprofile.health.HealthCheckResponse;

@Singleton
@Startup
public class FailingBean implements HealthCheck {

    @Override
    public HealthCheckResponse call() {
        return HealthCheckResponse.named("successful-check").up().build();
    }

    @PostConstruct
    public void init() {
        System.out.println("PostConstruct");
//        throw new RuntimeException();
    }
}
```

Based on this class, the named "successful-check" should appear, but it doesn't. Even setting the check to report a status of `down` doesn't cause any change. So I think there is a bug here, but it's a little more subtle than the one reported.

**Comment by smillidge** [comment]
Fixed by #3632
